LY to O3DE Migration Documents (#149)
o3de/community: Migration Guides/Lumberyard to O3DE/Misc-Documents/Keystone-Slice-Converter/README.md

> # Keystone Slice Converter

Reviewer: Isn't Keystone an internal code name? Should we avoid using that name in public?

> As a part of the Keystone effort, we're developing a slice-to-prefab (and level-to-prefab) converter.  This tool is intended to be short-lived, providing a one-time migration for a few white-glove customers.  Once the slice code is removed, this tool will no longer function and should be discarded.

Reviewer: This description should not mention white-glove customers
```suggestion
A slice converter tool exists to convert Lumberyard slices and levels into O3DE prefabs. This tool is intended to be short-lived and will be discarded after slice code is removed from O3DE.
```

To use, open a command line and enter a line like the following:

    SerializeContextTools.exe convert-slice --specializations=editor --project-path=D:\github\o3de\AutomatedTesting --files=D:\github\o3de\AutomatedTesting\Levels\*.ly

Usage Notes:

- Slices convert the same way, just with ".slice" instead of ".ly"
- Wildcards are recursive, so it will match any nested subdirectory
- Wildcards are optional - you can convert a single file by specifying it directly.

Scary output:

The following warnings / errors are all expected, and don't mean that something has gone wrong.

 <table>
  <tr> <td>==================================================================
Trace::Warning
 D:\github\o3de\Code\Framework\AzCore\AzCore\Settings\SettingsRegistryMergeUtils.cpp(416): 'bool __cdecl AZ::SettingsRegistryMergeUtils::MergeSettingsToRegistry_ConfigFile(class AZ::SettingsRegistryInterface &,class AZStd::basic_string_view<char,struct AZStd::char_traits<char> >,const struct AZ::SettingsRegistryMergeUtils::ConfigParserSettings &)'
Unable to open file "D:\github\o3de\AutomatedTesting\Cache\pc\user.cfg"
==================================================================</td>
  </tr>
</table> 

> GridMate Allocator has already started! Ignoring current allocator descriptor!

Reviewer: No GridMate output should occur anymore because it has been removed.

 - Standalone slices that have entity references "outside" of the slice data.  These simply can't be turned into equivalent prefab references without additional context in the converter as to where those references are intended to reference. 
   - One known cause of this is by using Trackview, following steps similar to the following: 
     - Create an entity.  Add a camera component to it.
     - Save the entity as a slice/prefab.
     - Open TrackView
     - Create new sequence
     - In Entity Inspector, select the camera entity
     - In TrackView, right-click and add selected entity
     - Save slice/prefab
     - Save level
 - Slice instances that have entity references that point "forward" to other slice instances that haven't been converted yet.
 - The serialized information about the component order is incorrect. See this PR for what the correct serialized form should look like : https://github.com/o3de/o3de/pull/8146/files

>  - Levels having entities using SCs might be broken after being converted by the tool if the SCs are having nodes with assigned entity IDs(For example: GetWorldTranslation, GetWorldTransform...). See ticket

Reviewer: ```suggestion
 - Levels having entities using Script Canvases might be broken after being converted by the tool if the Script Canvases have nodes with assigned entity IDs(For example: GetWorldTranslation, GetWorldTransform...).
```

> **Error rendering macro 'jira'**
> null

Reviewer: Remove any comments or text referring to internal Jira
```suggestion

```
